app/src/main/java/com/doctor/yumyum/presentation/ui/nickname/NicknameActivity.kt

```diff
@@ -0,0 +1,118 @@
+            binding.nicknameEtNickname.setText(nickname)
+        }
+        initNickname()
+    }
+
+    private fun init() {
+        binding.apply {
+            viewModel = viewModel
+            lifecycleOwner = this@NicknameActivity
+        }
+
+        binding.nicknameBtnComplete.setOnClickListener {
+            CoroutineScope(Dispatchers.IO).launch {
+                viewModel.patchNickname(binding.nicknameEtNickname.text.toString())
+            }
+        }
+
+        binding.nicknameEtNickname.addTextChangedListener(object : TextWatcher {
+            override fun beforeTextChanged(p0: CharSequence?, p1: Int, p2: Int, p3: Int) {}
+            override fun onTextChanged(p0: CharSequence?, p1: Int, p2: Int, p3: Int) {}
+
+            override fun afterTextChanged(p0: Editable?) {
+                CoroutineScope(Dispatchers.IO).launch {
+                    val unique = viewModel.validateNickname(p0.toString())
+                    if (!unique) {
+                        setMessageOverlap()
+                        setButtonUnavailable()
+                    }
+                }
+                if (p0.isNullOrEmpty()) {
+                    setMessageNull()
+                    setButtonUnavailable()
+                } else {
+                    if (p0.length >= 20) {
+                        setMessageOverflow()
+                        setButtonUnavailable()
+                    } else {
+                        setMessageSuccess()
+                        setButtonAvailable()
+                    }
+                }
+            }
+        })
+    }
+
+    private fun initNickname() {
+        CoroutineScope(Dispatchers.IO).launch {
+            viewModel.getNickname()
+        }
+    }
+
+    fun setMessageNull() {
+        binding.nicknameTvMessage.visibility = View.INVISIBLE
+    }
+
+    fun setMessageSuccess() {
+        binding.nicknameTvMessage.visibility = View.VISIBLE
+        binding.nicknameTvMessage.setTextColor(getColor(R.color.sub_green))
+        binding.nicknameTvMessage.text = getString(R.string.nickname_tv_success)
+    }
+
+    fun setMessageOverlap() {
+        binding.nicknameTvMessage.visibility = View.VISIBLE
+        binding.nicknameTvMessage.setTextColor(getColor(R.color.main_orange))
+        binding.nicknameTvMessage.text = getString(R.string.nickname_tv_overlap)
+    }
+
+    fun setMessageOverflow() {
+        binding.nicknameTvMessage.visibility = View.VISIBLE
+        binding.nicknameTvMessage.setTextColor(getColor(R.color.main_orange))
+        binding.nicknameTvMessage.text = getString(R.string.nickname_tv_overflow)
+    }
+
+    fun setButtonUnavailable() {
+        binding.nicknameBtnComplete.apply {
+            isClickable = false
+            background = getDrawable(R.drawable.bg_btn_sub)
+        }
+    }
+
+    fun setButtonAvailable() {
+        binding.nicknameBtnComplete.setBackgroundResource(R.drawable.bg_btn_main)
+        binding.nicknameBtnComplete.isClickable = true
+    }
```

Comment:
두 부분 다 apply를 쓰거나 안쓰는 식으로 통일하면 좋을 것 같아요.
```suggestion
    fun setButtonUnavailable() {
        binding.nicknameBtnComplete.setBackgroundResource(R.drawable.bg_btn_sub)
        binding.nicknameBtnComplete.isClickable = false
    }

    fun setButtonAvailable() {
        binding.nicknameBtnComplete.setBackgroundResource(R.drawable.bg_btn_main)
        binding.nicknameBtnComplete.isClickable = true
    }
```